webview_panels/src/modules/defer/DeferToProduction.tsx

```diff
@@ -93,23 +125,102 @@ const DeferToProduction = (): JSX.Element => {
     }
   };
 
-  const handleManifestPathChange = (
+  const handleLocalManifestPathChange = (
     event: React.ChangeEvent<HTMLInputElement>,
   ) => {
     const { value, name } = event.target;
+    if (!value.endsWith("manifest.json")) {
+      setShowManifestError(false);
+    } else {
+      setShowManifestError(true);
+    }
     setDeferState((prevState) => ({ ...prevState, [name]: value }));
   };
 
-  const onManifestBlur = async () => {
-    const response = await executeRequestInSync("updateDeferConfig", {
-      key: "manifestPathForDeferral",
-      value: manifestPathForDeferral,
-      isPreviewFeature: true,
+  const onLocalManifestBlur = async () => {
+    if (!showManifestError) {
+      const response = await executeRequestInSync("updateDeferConfig", {
+        key: "manifestPathForDeferral",
+        value: manifestPathForDeferral,
+        isPreviewFeature: true,
+      });
+      if (!(response as { updated: boolean }).updated) {
+        setDeferState((prevState) => ({
+          ...prevState,
+          manifestPathForDeferral: "",
+        }));
+      }
+
+      const response2 = await executeRequestInSync("updateDeferConfig", {
+        key: "manifestPathType",
+        value: manifestPathType,
+      });
+      if (!(response2 as { updated: boolean }).updated) {
+        setDeferState((prevState) => ({
+          ...prevState,
+          manifestPathType: "",
+        }));
+      }
+    }
+  };
+
+  const setProjectIntegrations = async () => {
+    const response = await executeRequestInSync("fetchProjectIntegrations", {});
+    if (Array.isArray(response)) {
+      setDeferState((prevState) => ({
+        ...prevState,
+        projectIntegrations: response.map(
+          (item: { name: string; id: number }) => ({
+            label: item.name,
+            value: item.id,
+          }),
+        ),
+        manifestPathType: "remote",
+      }));
+      return;
+    }
+  };
+
+  const handleManifestPathTypeChange = async (option: string) => {
+    if (option === "remote") {
+      await setProjectIntegrations();
+    }
+    setDeferState((prevState) => ({
+      ...prevState,
+      manifestPathType: option,
+    }));
+  };
+
+  const handleIntegrationSelect = async (selectedOption: {
+    label: string;
+    value: number;
+  }) => {
+    const response = await executeRequestInSync("testRemoteManifest", {
+      dbt_core_integration_id: selectedOption.value,
     });
-    if (!(response as { updated: boolean }).updated) {
+    const data = response as {
+      url: string;
+      dbt_core_integration_file_id: number;
+    };
+    if (data.url === "" && data.dbt_core_integration_file_id === -1) {
+      executeRequestInAsync("showInformationMessage", {
+        infoMessage: `No remote manifest file present for dbt core integration: ${selectedOption.label}`,
+      });
+    } else {
```

Comment:
you could do early `return` for the `if` condition and avoid else block to make the code more maintainable and readable
```
if (condition){
return
}
do next steps
```